lib/miq_automation_engine/engine/miq_ae_engine.rb

```diff
@@ -262,8 +262,15 @@ def self.create_automation_attribute_array_key(key)
   end
 
   def self.create_automation_attribute_array_value(value)
-    value.collect do |obj|
-      obj.kind_of?(ActiveRecord::Base) ? obj.id.to_s : obj.to_s
+    # When array of length 1 just return the single object or id instead of an array
+    if value.length == 1
+      obj = value[0]
+      obj.kind_of?(ActiveRecord::Base) ? "#{obj.class.name}::#{obj.id}" : obj.to_s
+    else
+      # Return an array of object references or ids
+      value.collect do |obj|
+        obj.kind_of?(ActiveRecord::Base) ? "#{obj.class.name}::#{obj.id}" : obj.to_s
+      end
```

Comment:
is this the same as:


```ruby
      result = value.collect do |obj|
        obj.kind_of?(ActiveRecord::Base) ? "#{obj.class.name}::#{obj.id}" : obj.to_s
      end
      result.length == 1 ? result.first : result
```

It looks like we want to do the same thing regardless if it's a 1 element array or more, but we want to return just the first result if it's a single element array.

Reply:
Do we need to be concerned with that we still are missing the `\x1F` that was there previously:  https://github.com/ManageIQ/manageiq-automation_engine/pull/545

I don't understand why that was there.

Reply:
The \x1F was first introduce for string arrays that could have _actual_ commas in the value.  We chose a character that is designed for being a field separate and should never actually be a string member (\x1F is the ASCII character for a "Unit separator")

Reply:
We don't need the \x1F in this case because we are not converting the array values into a string but instead keeping them in an actual array.

Reply:
I think it's fine to preserve the prior behavior.  It's odd that both the automate backend side and the user's automation scripts need to have conditional logic to deal with "sometimes array of strings, sometimes a single string"... but that's how it was.

I think we can deduplicate our code by building an array in the same way, then just return the first element of the array in the case of a 1 length array.  See my comment above.  I think that preserves the logic you had there.